### PR TITLE
#617 add get*Mapping and set*Mapping for users/roles/profiles

### LINKED ISSUE #617: Mapping on user/profile/role
Hello,

Because we can't perform a `.getMapping` on the index `%kuzzle` we can't display a proper form on the back office with right fields.
We need to have a way to get mapping for user/profile/role in Kuzzle and in SDKs.

### PR DESCRIPTION
**Pull request by ballinette**
**NB : need to wait merge of #616 before adding the last fix with asserts**

fix #617 
# Implemented actions
* move `getUserMapping` and `setUserMapping` actions form `collection` to `security` controller 
* add `getProfileMapping`, `getRoleMapping`, `setProfileMapping` and `setRoleMapping` actions

# Boy-scout 
* add `'use strict'` when missing
* refactor security controller unit-tests to use a Kuzzle Mock.

### CODE REVIEW

File: lib/api/controllers/securityController.js [hunk 1]
```diff
@@ -19,6 +19,68 @@ var
...
+
+  /**
+   * Get the profile mapping
+   *
+   * @returns {Promise}
+   */
+  this.getProfileMapping = function securityGetProfileMapping () {
+    return kuzzle.internalEngine.getMapping({index: kuzzle.internalEngine.index, type: 'profiles'})
+      .then(response => {
+        return Promise.resolve({mapping: response[kuzzle.internalEngine.index].mappings.profiles.properties});
```
Comment (AnthonySendra) [inline lib/api/controllers/securityController.js]: Why not just `return {mapping: response[kuzzle.internalEngine.index].mappings.profiles.properties};`?

File: features/HTTP.feature [hunk 2]
```diff
@@ -170,6 +170,15 @@ Feature: Test HTTP API
     Then I create a restricted user "restricteduser1" with id "restricteduser1-id"
 
   @usingHttp @cleanSecurity
+  Scenario: Role mapping
+    Given I get the role schema
```
Comment (xbill82) [inline features/HTTP.feature]: I have the feeling there is ambiguity between "schema" and "mapping". Do we really need both terms?

**Comment by xbill82** [comment]
Needs to fix tests. Not merging now.

File: features/HTTP.feature (hunk 2 above)
Comment (ballinette) [inline features/HTTP.feature]: Nope, I just kept the term from tests on `getMapping()`

Comment (ballinette) [inline features/HTTP.feature]: @xbill82  I just renamed every "schema" term instance by "mapping" ;)

File: lib/api/controllers/securityController.js [hunk 3]
```diff
@@ -19,6 +19,66 @@ var
...
+      .then(response => ({mapping: response[kuzzle.internalEngine.index].mappings.roles.properties}));
+  };
+
+  /**
+   * Update the roles collection mapping
+
+   * @param {Request} request
+   * @returns {Promise}
+   */
+  this.updateRoleMapping = function securityUpdateRoleMapping (request) {
```
Comment (dbengsch) [inline lib/api/controllers/securityController.js]: It could use `assertHasBody` (same for other update*Mapping methods)

Comment (ballinette) [inline lib/api/controllers/securityController.js]: Yep, I'm waiting the merge of you're PR #616 because you changed the `assert*` signatures there

Comment (ballinette) [inline lib/api/controllers/securityController.js]: done

**Pull request merged**